website/docs/generics.md

```diff
@@ -0,0 +1,1102 @@
+---
+id: generics
+title: Generic Classes and Methods
+sidebar_label: Generics
+---
+
+Sorbet has syntax for creating generic methods, classes, and interfaces.
+
+## Implementation status
+
+Generics are the most poorly supported feature within Sorbet, by far. Their
+implementation has many known, complicated-to-fix bugs which substantially limit
+their utility.
+
+It is hard to overstate how buggy Sorbet's support for generics is.
+
+We track known bugs in Sorbet's support for generics in the [Generics milestone]
+on Sorbet's issue tracker. Feel free to peruse the existing bugs, as well as
+report new bugs.
+
+Unfortunately, this abundance of bugs makes implementing sophisticated, generic
+abstractions in Sorbet exceedingly brittle.
+
+It is imperative to thoroughly test abstractions making use of Sorbet generics.
+
+The tests you'll need to write look substantially different from other Ruby
+tests you may be accustomed to writing, because they will have to live **at the
+type level** (statics), not at the expression level (runtime). These tests
```

Comment:
I think "type level" vs. "expression level" is a bit too much for this documentation.  I don't know if the below is a better way to phrase it, but it at least points in the right direction.

```suggestion
The tests you'll need to write look substantially different from other Ruby
tests you may be accustomed to writing, because the tests need to deal with
what code should or should not typecheck, rather that code that should or
should not run correctly.   These tests
```